### ISSUE TITLE
Improve tap

### ISSUE THREAD

**Issue by thomascoumau** [OwnWeb/flutter_map_tappable_polyline#59]
Hello, I have already made a PR on this repo to improve passing the tap to the map and I have been working on improving the polyline tap for some month. I have made a PR on flutterMap [here](https://github.com/fleaflet/flutter_map/pull/1686), however the maintainer prefer to let this responsibility to this pluggin. I can rework my PR on FlutterMap repo a little to reopen it on this pluggin if you would like to.

**Comment by HugoHeneault** [comment]
Yes it would be great! Would you consider waiting for #60 to be merged or base your work on it?

Thanks alot! :)

**Comment by thomascoumau** [comment]
I can wait for #60 to be merged of course.

However in my PR I reworked completely how the package is working. I decided to make a custom paint for each polyline instead of the layer and to implement on override on their hitTest with a touchable path. 

This prevent having to track where on the map the polyline is tapped and also it prevent from having to bother on how to pass the tap on the map below the polylines.

**Comment by alexbejann** [comment]
Hey @thomascoumau, did you try to implement these [changes](https://github.com/fleaflet/flutter_map/pull/1686) into the package?
